quickwit/quickwit-aws/src/lib.rs

```diff
@@ -19,94 +19,55 @@
-    let builder = HttpsConnectorBuilder::new();
-    let builder = builder.with_native_roots();
-    let connector = builder
-        .https_or_http()
-        // We do not enable HTTP2.
-        // It is not enabled on S3 and it does not seem to work with Google Cloud Storage at
-        // this point. https://github.com/quickwit-oss/quickwit/issues/1584
-        //
-        // (HTTP2 would be awesome since we do a lot of concurrent requests and
-        // HTTP2 enables multiplexing a given connection.)
-        .enable_http1()
-        .build();
-    HttpClient::from_connector_with_config(connector, http_config)
+/// Attempts to get the current AWS SDK config.
+///
+/// If the config has not yet been initialised this will return `None`.
+pub fn try_get_aws_config() -> Option<&'static aws_config::SdkConfig> {
+    AWS_CONFIG.get()
 }
 
-/// Returns a singleton credentials provider.
-pub fn get_credentials_provider() -> anyhow::Result<impl ProvideAwsCredentials + Send + Sync> {
-    static CREDENTIALS_PROVIDER_SINGLETON: OnceCell<AutoRefreshingProvider<ExtendedChainProvider>> =
-        OnceCell::new();
-    CREDENTIALS_PROVIDER_SINGLETON
-        .get_or_try_init(move || {
-            let chain_provider = ExtendedChainProvider::new();
-            let credentials_provider = AutoRefreshingProvider::new(chain_provider)
-                .context("Failed to instantiate AWS credentials provider.")?;
-            Ok(credentials_provider)
-        })
-        .cloned()
-}
+/// Attempts to initialise the AWS SDK config.
+///
+/// If the config is already initialised, this is a no-op.
+pub async fn try_init_aws_config() -> &'static aws_config::SdkConfig {
+    AWS_CONFIG
+        .get_or_init(|| async move {
+            let builder = HttpsConnectorBuilder::new();
+            let builder = builder.with_native_roots();
+            let connector = builder
+                .https_or_http()
+                // We do not enable HTTP2.
+                // It is not enabled on S3 and it does not seem to work with Google Cloud Storage at
+                // this point. https://github.com/quickwit-oss/quickwit/issues/1584
+                //
+                // (HTTP2 would be awesome since we do a lot of concurrent requests and
+                // HTTP2 enables multiplexing a given connection.)
+                .enable_http1()
+                .build();
 
-#[derive(Clone, Debug)]
-struct ExtendedChainProvider {
-    chain_provider: ChainProvider,
-    web_identity_provider: WebIdentityProvider,
-}
+            let smithy_connector = hyper_ext::Adapter::builder().build(connector);
 
-impl ExtendedChainProvider {
-    fn new() -> Self {
-        let mut chain_provider = ChainProvider::new();
-        chain_provider.set_timeout(CREDENTIALS_FETCH_TIMEOUT);
-        let web_identity_provider = WebIdentityProvider::from_k8s_env();
-        Self {
-            chain_provider,
-            web_identity_provider,
-        }
-    }
+            aws_config::from_env()
```

Comment:
Did you check if it covers the k8s case? (it seems so but just want you to double check). We could add a task such as "test k8s credentials on k8s cluster" in the to-do list.

Reply:
I haven't checked, what system does it normally use? (I.e. ENV vars, etc...)

Reply:
I think you work on it in the PR related to credentials errors (web identity provider etc.). But the defaults seem quite nice so hopefully we don't have much to do there.